### PR TITLE
[expo-auth-session] Fix release channel support

### LINKED ISSUE #7674: expo-auth-session does not support release channels due to incorrect defaultReturnUrl
## 🐛 Bug Report

expo-auth-session defaultReturnUrl is incorrect if you are using release channels. You can not simply append a string to a url that has a query parameter in it.

### Environment

  Expo CLI 3.17.15 environment info:
    System:
      OS: Windows 10 10.0.17763
    Binaries:
      Node: 12.6.0 - C:\Program Files\nodejs\node.EXE
      npm: 6.14.4 - C:\Program Files\nodejs\npm.CMD
    IDEs:
      Android Studio: Version  3.5.0.0 AI-191.8026.42.35.6010548
    npmPackages:
      expo: ^37.0.0 => 37.0.4
      react: 16.9.0 => 16.9.0
      react-native: https://github.com/expo/react-native/archive/sdk-37.0.0.tar.gz => 0.61.4

all

### Steps to Reproduce

`expo publish --release-channel releases-v0.3`
Try and log in to the app

### Expected Behavior

Login returns  to ` exp://exp.host/@{user}/{project}/--/expo-auth-session?release-channel=releases-v0.3`

### Actual Behavior

Login attempts to returns  to ` exp://exp.host/@{user}/{project}?release-channel=releases-v0.3/--/expo-auth-session` and fails

### Workaround

You can workaround the issue until a patch is made by splitting the default return url into pieces and reconstructing the correct url as follows:

```
      let returnUrl = AuthSession.getDefaultReturnUrl()
      let returnUrlParts = returnUrl.split('?')
      if (returnUrlParts.length > 1) {
        let extra = returnUrlParts[1].split('/')
        let queryParams = extra.shift()
        returnUrl = `${returnUrlParts[0]}/${extra.join('/')}?${queryParams}`
      }
      let result = await (AuthSession.startAsync({authUrl: url, returnUrl: returnUrl }))
```

Verified this works using https://expo.io/@celandro/raid-party?release-channel=releases-v0.3

### PR DESCRIPTION
**Pull request by lukmccall**
# Why

Resolves https://github.com/expo/expo/issues/7674.

# How

Fixes `AuthSession.getDefaultReturnUrl()` 

# Test Plan

- run this code in unpublished and published app: 
```ts
React.useEffect(() => {
    const f = async () => {
      alert(AuthSession.getDefaultReturnUrl());
    };
    f();
  }, []);
```
- run `adb shell am start -d <result of AuthSession.getDefaultReturnUrl()>` (for me: "exp://exp.host/@lukaszkosmaty/sandbox/--/expo-auth-session?release-channel=releases-test")

### CODE REVIEW

**Comment by brentvatne** [comment]
@lukmccall - can you write a test to cover this case?

File: packages/expo-auth-session/CHANGELOG.md [hunk 1]
```diff
@@ -7,3 +7,5 @@
 ### 🎉 New features
 
 ### 🐛 Bug fixes
+
+- Fix `AuthSession.getDefaultReturnUrl()` returning wrong URL while using releas channels. ([#7687](https://github.com/expo/expo/pull/7687) by [@lukmccall](https://github.com/lukmccall))
```
Comment (bbarthec) [inline packages/expo-auth-session/CHANGELOG.md:11]:
```suggestion
- Fix `AuthSession.getDefaultReturnUrl()` returning wrong URL while using release channels. ([#7687](https://github.com/expo/expo/pull/7687) by [@lukmccall](https://github.com/lukmccall))
```

**Review by bbarthec — APPROVED** [review]

**Review by tsapeta — APPROVED** [review]
Approved, but would be great to write a test as Brent suggested 😉

**Comment by brentvatne** [comment]
this is available in `expo-auth-session@1.0.1`

**Pull request merged**